lib/ClusterShell/Gateway.py

```diff
@@ -343,7 +343,7 @@ def gateway_main():
     task.set_default("stdout_msgtree", False)
     task.set_default("stderr_msgtree", False)
 
-    if sys.stdin.isatty():
+    if sys.stdin is not None and sys.stdin.isatty():
```

Comment:
I would prefer to split this into two separate conditions, and update the error message accordingly in case sys.stdin is None.

Reply:
Is the gateway code guaranteed to always have stdin?

Reply:
I see it uses it later, so seems like a fair assumption.

Reply:
Good point. The gateway module is launched by clustershell itself and stdin is always connected, so it's very unlikely that sys.stdin would be None. So maybe there is nothing to change here after all. if sys.stdin is None here, it will fail anyway and we can't do anything about it. So maybe do nothing?